src/backend/commands/createas.c

```diff
@@ -609,28 +419,239 @@ ExecCreateTableAs(CreateTableAsStmt *stmt, const char *queryString,
 
 		if (into->ivm)
 		{
-
 			Oid matviewOid = address.objectId;
 			Relation matviewRel = table_open(matviewOid, NoLock);
 			Relids	relids = NULL;
 
-			copied_query = copyObject(query);
-			AcquireRewriteLocks(copied_query, true, false);
+			/*
+			 * Mark relisivm field, if it's a matview and into->ivm is true.
+			 */
+			SetMatViewIVMState(matviewRel, true);
```

Comment:
Is there a reason for moving SetMatViewIVMState here?

Reply:
The before place was not called, if we use WITH NO DATA option. So I moved it for supporting WITH NO DATA option.

Reply:
Thanks!  I agree with this change.